### PR TITLE
Add support for long to ToQuantity

### LINKED ISSUE #503: ToQuantity does not support long quantities
It would be useful if ToQuantity supported long quantities in addition to int. Something like the following would work but will not honor ShowQuantityAs.Words for large positive or negative quantity values:

``` csharp
public static string ToQuantity(this string input, long quantity, ShowQuantityAs showQuantityAs = ShowQuantityAs.Numeric, string format = null, IFormatProvider formatProvider = null)
{
    var str = quantity == 1 ? input.Singularize(false) : input.Pluralize(false);
    if (showQuantityAs == ShowQuantityAs.None)
        return str;
    if ((showQuantityAs == ShowQuantityAs.Numeric) || (quantity > int.MaxValue) || (quantity < int.MinValue))
        return string.Format(formatProvider, "{0} {1}", quantity.ToString(format, formatProvider), str);
    return string.Format("{0} {1}", ((int) quantity).ToWords(), str);
}
```

### PR DESCRIPTION
**Pull request by RobPethick**
Fixes #503

---

This change is [Reviewable](https://reviewable.io/reviews/humanizr/humanizer/504)

### CODE REVIEW

**Comment by mexx** [comment]
Sorry, but what's the purpose to extend the domain range by allowing `long` but throw exceptions when this extended range is used? The right fix would be to provide `ToWords` for `long`, also see my [comment](https://github.com/Humanizr/Humanizer/issues/503#issuecomment-166312494)

**Comment by RobPethick** [comment]
@mexx sorry I hadn't seen your comment before putting up my PR. Obviously the point is to allow for it for ShowQuantityAs.None and ShowQuantityAs.Numeric. But I appreciate it kind of seems half complete.

**Comment by RobPethick** [comment]
added in ShowQuantityAs.Words

File: src/Humanizer.v2.ncrunchsolution [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<SolutionConfiguration>
```
Comment (mexx) [inline src/Humanizer.v2.ncrunchsolution]: Is this file needed? Isn't it only a configuration for NCrunch?

Comment (RobPethick) [inline src/Humanizer.v2.ncrunchsolution]: good catch, I have removed it

**Comment by mexx** [comment]
I don't know should we mention this in the `readme.md` somehow?

**Comment by RobPethick** [comment]
I could, though in general do we discuss types in the readme?

**Comment by mexx** [comment]
You are right, there is no actual notion of the types in the readme.

LGTM, @onovotny can you have a look also and merge if it's all OK?

**Comment by onovotny** [comment]
Rebased and merged here: https://github.com/Humanizr/Humanizer/commit/e88b07ee6a31aa7a0ed1f59e51a1d0edc1b7865a, thanks!
